internal/core/metadata/rest_addressable_test.go

```diff
@@ -53,8 +53,7 @@ var ErrorPathParam = "%zz"
 var ErrorPortPathParam = "abc"
 
 func TestMain(m *testing.M) {
-	LoggingClient = logger.NewMockClient()
-	httpErrorHandler = errorconcept.NewErrorHandler(LoggingClient)
+	httpErrorHandler = errorconcept.NewErrorHandler(logger.NewMockClient())
 	os.Exit(m.Run())
 }
```

Comment:
`TestMain` doesn't actually test anything and only exists, I think, to goose code coverage stats.  Please remove it.  This is consistent with the equivalent/corresponding work done on other services.

Reply:
- It turns out that removal of `TestMain` leads to the following error during `make test`, so I have retained it for now:
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1588c62]

goroutine 23 [running]:
testing.tRunner.func1(0xc00016ce00)
        /usr/local/go/src/testing/testing.go:830 +0x392
panic(0x1671920, 0x1cbbd50)
        /usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/edgexfoundry/edgex-go/internal/core/metadata.restGetAllAddressables(0x184e720, 0xc000129300, 0x185a800, 0x1ceebb0)
        /Users/akramahmad/go/src/github.com/edgexfoundry/edgex-go/internal/core/metadata/rest_addressable.go:40 +0x422
github.com/edgexfoundry/edgex-go/internal/core/metadata.TestGetAllAddressables.func2(0xc00016ce00)
        /Users/akramahmad/go/src/github.com/edgexfoundry/edgex-go/internal/core/metadata/rest_addressable_test.go:92 +0xf2
testing.tRunner(0xc00016ce00, 0xc000133360)
```
where, Line 40:
    httpErrorHandler.HandleOneVariant(w, err, errorconcept.Common.LimitExceeded, errorconcept.Default.InternalServerError)

and, Line 92:
      restGetAllAddressables(rr, logger.NewMockClient())

Reply:
To remove `TestMain()`, you need to add this line to every test until you remove the `httpErrorHandler` global:

`httpErrorHandler = errorconcept.NewErrorHandler(logger.NewMockClient())`

Reply:
Discussed out-of-band on Slack; request is to remove hidden dependency on TestMain so you can remove it as part of this PR.

Reply:
- Yes, the next step here is for me to figure out the hidden dependencies some tests have on `TestMain` and fulfill those dependencies within each test that has them so I can remove `TestMain` 👍

Reply:
@michaelestrin Once I have addressed this comment as well, I will notify (via `Request Review` option up atop) at that time that this PR is ready again for your review.